lib/cn/route.c

```diff
@@ -210,30 +81,18 @@ route_node_key_set(struct route_node *node, const void *edge_key, uint edge_klen
 struct route_node *
 route_node_alloc(struct route_map *map, void *tnode, const void *edge_key, uint edge_klen)
 {
-    struct route_node *node = map->rtm_free;
-    size_t nodesz;
-
-    if (node) {
-        merr_t err = route_node_keybuf_alloc(node, edge_klen);
-        if (err)
-            return NULL;
-
-        map->rtm_free = node->rtn_next;
-    } else {
-        /* The route_node cache is empty, allocate a new route_node.
-         * These allocated route_node entries are never cached.
-         */
-        nodesz = sizeof(*node);
-        if (edge_klen > sizeof(node->rtn_keybuf))
-            nodesz += ALIGN(edge_klen, __alignof__(*node));
-
-        node = aligned_alloc(__alignof__(*node), nodesz);
-        if (!node)
-            return NULL;
-
-        memset(node, 0, nodesz);
-        node->rtn_keybufp = (nodesz > sizeof(*node)) ? (uint8_t *)(node + 1) : node->rtn_keybuf;
-    }
+    struct route_node *node;
+    merr_t err;
+
+    node = map->rtm_free;
+    if (!node)
+        return NULL;
+
+    err = route_node_keybuf_alloc(node, edge_klen);
+    if (err)
```

Comment:
Swallowing the error looks weird here, even though the only failure is ENOMEM. Would it make a little more sense to return a boolean indicating whether the memory was allocated or not?

Reply:
As the only failure path is ENOMEM, I think it is fine the way it is.

Reply:
Yea but you have to go read the function to know that that is the only failure which isn't ideal

Reply:
Sure, but then I've rarely found it useful to know precisely why, for example, malloc() failed...